### ISSUE TITLE
Error installing quanteda.corpora package on Windows laptop

### ISSUE THREAD

**Issue by BasBijlsma** [quanteda/quanteda.corpora#7]
All,  
When installing the package I get the below error. Tried to solve together with my LAB-instructor for the LSE ME314 summer school, but we didn't manage to. Any suggestions?  

I saw in another post that (re-)installing RTools could help, but it didn't solve it for me. Am working from the server now but would like to use the package on my own computer in the future. 

Thank you! 

Bas

Downloading GitHub repo quanteda/quanteda.corpora@master
from URL https://api.github.com/repos/quanteda/quanteda.corpora/zipball/master
Installing quanteda.corpora
"C:/PROGRA~1/R/R-35~1.1/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD  \
  INSTALL "C:/Users/Co�rdinator  \
  HIL/AppData/Local/Temp/RtmpELJW1H/devtools239092d792c/quanteda-quanteda.corpora-91a3674"  \
  --library="C:/Users/Co�rdinator HIL/Documents/R/win-library/3.5" --install-tests 

* installing *source* package 'quanteda.corpora' ...
Warning in file(file, if (append) "a" else "w") :
  cannot open file 'C:/Users/Covrdinator HIL/Documents/R/win-library/3.5/quanteda.corpora/DESCRIPTION': No such file or directory
Error in file(file, if (append) "a" else "w") : 
  cannot open the connection
ERROR: installing package DESCRIPTION failed for package 'quanteda.corpora'
* removing 'C:/Users/Coördinator HIL/Documents/R/win-library/3.5/quanteda.corpora'
In R CMD INSTALL
Installation failed: Command failed (1)

**Comment by kbenoit** [comment]
Please supply:

* the output from `sessionInfo()`
* the command you used to (attempt to) install the package

**Comment by BasBijlsma** [comment]
Here you go:

**#session_info()**

Session info --------------------------------------------------------------------------------------

Packages ------------------------------------------------------------------------------------------
 package   * version date       source        
 base      * 3.5.1   2018-07-02 local         
 compiler    3.5.1   2018-07-02 local         
 curl        3.2     2018-03-28 CRAN (R 3.5.1)
 datasets  * 3.5.1   2018-07-02 local         
 devtools  * 1.13.6  2018-06-27 CRAN (R 3.5.1)
 digest      0.6.15  2018-01-28 CRAN (R 3.5.1)
 git2r       0.23.0  2018-07-17 CRAN (R 3.5.1)
 graphics  * 3.5.1   2018-07-02 local         
 grDevices * 3.5.1   2018-07-02 local         
 httr        1.3.1   2017-08-20 CRAN (R 3.5.1)
 knitr       1.20    2018-02-20 CRAN (R 3.5.1)
 memoise     1.1.0   2017-04-21 CRAN (R 3.5.1)
 methods   * 3.5.1   2018-07-02 local         
 R6          2.2.2   2017-06-17 CRAN (R 3.5.1)
 stats     * 3.5.1   2018-07-02 local         
 tools       3.5.1   2018-07-02 local         
 utils     * 3.5.1   2018-07-02 local         
 withr       2.1.2   2018-03-15 CRAN (R 3.5.1)

**the command used to (attempt to) install the package**

devtools::install_github('quanteda/quanteda.corpora')

**Comment by kbenoit** [comment]
I just verified the installation and it worked fine. I suspect that either you lack permissions to write to that installation folder, or are experiencing some network connection problem. (It's a fairly large package because of the size of the data objects.)

Also your `sessionInfo()` output should look something like:

```r
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.5.1
```

**Comment by BasBijlsma** [comment]
Could it have to do with the fact that the name of the installation folder includes an 'ö', namely in 'Coördinator'? In the error message it is written as 'Co�rdinator' and 'Covrdinator'... Is the installation folder easily changed? Or should I reinstall R for this? Google doesn't provide a clear answer to this.. 

I have now pasted the return of the correct function `sessionInfo() ` below (I had used a different function). It is quite different from what you've posted above. E.g. under 'locale' yours references to UTF-8 etc but mine to some things related to my national language. As you can tell I am totally new to this, but could this be the problem? I didn't have any problems installing packages before, but it would be typical for a text analysis package to find this a problem, wouldn't it?! :-)

This is the return I get from sessionInfo():
```
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=Dutch_Netherlands.1252  LC_CTYPE=Dutch_Netherlands.1252   
[3] LC_MONETARY=Dutch_Netherlands.1252 LC_NUMERIC=C                      
[5] LC_TIME=Dutch_Netherlands.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] devtools_1.13.6

loaded via a namespace (and not attached):
 [1] httr_1.3.1     compiler_3.5.1 R6_2.2.2       tools_3.5.1    withr_2.1.2    rstudioapi_0.7
 [7] curl_3.2       memoise_1.1.0  knitr_1.20     git2r_0.23.0   digest_0.6.15

```

**Comment by kbenoit** [comment]
If that were it, you would have trouble installing any package. Try installing something else from GitHub, e.g.
```r
devtools::install_github("quanteda/stopwords")
```

**Comment by BasBijlsma** [comment]
Returns same error:

```
> devtools::install_github("quanteda/stopwords")
Downloading GitHub repo quanteda/stopwords@master
from URL https://api.github.com/repos/quanteda/stopwords/zipball/master
GitHub repo contains submodules, may not function as expected!Installing stopwords
"C:/PROGRA~1/R/R-35~1.1/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD  \
  INSTALL "C:/Users/Co�rdinator  \
  HIL/AppData/Local/Temp/RtmpWksfzb/devtools4e2c2e6969f1/quanteda-stopwords-ea88a3f"  \
  --library="C:/Users/Co�rdinator HIL/Documents/R/win-library/3.5" --install-tests 

* installing *source* package 'stopwords' ...
Warning in file(file, if (append) "a" else "w") :
  cannot open file 'C:/Users/Covrdinator HIL/Documents/R/win-library/3.5/stopwords/DESCRIPTION': No such file or directory
Error in file(file, if (append) "a" else "w") : 
  cannot open the connection
ERROR: installing package DESCRIPTION failed for package 'stopwords'
* removing 'C:/Users/Coördinator HIL/Documents/R/win-library/3.5/stopwords'
* restoring previous 'C:/Users/Coördinator HIL/Documents/R/win-library/3.5/stopwords'
In R CMD INSTALL
Installation failed: Command failed (1)
```

**Comment by kbenoit** [comment]
Does this work?
```r
install.packages("stopwords")
```

And also try this:
```r
devtools::install_github("mvuorre/exampleRPackage")
```

If either fails, then you have a permissions or configuration problem with R, and might need to reinstall, or change your user name. R installs packages by default into 
```r
> .libPaths()[1]
[1] "/Users/kbenoit/Library/R/3.5/library"
```
and it could be that in your case, Windows can't handle the Unicode character in your user name, so it's mangling "Coördinator". (Welcome to Unicode on Windows.)

It that fails too, then it's definitely.

**Comment by kbenoit** [comment]
Note that when you paste code blocks, it should be:
````
```{r}
# code here
``` 
````

**Comment by BasBijlsma** [comment]
Thanks for your help, Ken. I have changed the library directory and it seems to work now. I have just succesfully installed and used the packages. 

I used the below suggestion to change the library directory:

_The proper solution is to set environment variable R_LIBS_USER to the value of the file path to your desired library folder as opposed to getting RStudio to recognize a Rprofile.site file.

To set environment variable R_LIBS_USER in Windows, go to the Control Panel (System Properties -> Advanced system properties -> Environment Variables -> User Variables) to a desired value (the path to your library folder), e.g._

```
Variable name: R_LIBS_USER 
Variable value: C:/Rlibs  
```
_Setting R_LIBS_USER will ensure that the library shows up first in .libPaths() regardless of starting RStudio directly or by right-clicking an file and "Open With" to start RStudio.

The Rprofile solution can work if RStudio is always started by clicking the RStudio shortcut. In this case, setting the default working directory to the directory that houses your Rprofile will be sufficient. The Rprofile solution does not work when clicking on a file to start RStudio because that changes the working directory away from the default working directory._

SOURCE: [https://stackoverflow.com/questions/15170399/changing-r-default-library-path-using-libpaths-in-rprofile-site-fails-to-work](url)
